examples/py_triple.rs

```diff
@@ -0,0 +1,411 @@
+            let r = cs.query_advice(r_col, Rotation::cur());
+            let o = cs.query_advice(o_col, Rotation::cur());
+            let s_add = cs.query_selector(s_add, Rotation::cur());
+            s_add * (l + r - o)
+        });
+
+        // Define the multiplication gate.
+        //
+        // | l_col | r_col | o_col | s_mul |
+        // |-------|-------|-------|-------|
+        // |   l   |   r   |   o   | s_mul |
+        //
+        // Constraint: s_mul*l*r = s_mul*o
+        cs.create_gate("mul", |cs| {
+            let l = cs.query_advice(l_col, Rotation::cur());
+            let r = cs.query_advice(r_col, Rotation::cur());
+            let o = cs.query_advice(o_col, Rotation::cur());
+            let s_mul = cs.query_selector(s_mul, Rotation::cur());
+            s_mul * (l * r - o)
+        });
+
+        // Define the public input gate.
+        //
+        // | l_col | pub_col | s_pub |
+        // |-------|---------|-------|
+        // |   l   |    pi   | s_pub |
+        //
+        // Constraint: s_pub*l = s_pub*pi
+        cs.create_gate("public input", |cs| {
+            let l = cs.query_advice(l_col, Rotation::cur());
+            let pi = cs.query_instance(pub_col, Rotation::cur());
+            let s_pub = cs.query_selector(s_pub, Rotation::cur());
+            s_pub * (l - pi)
+        });
+
+        MyChipConfig {
+            l_col,
+            r_col,
+            o_col,
+            perm,
+            s_add,
+            s_mul,
+            s_pub,
+        }
+    }
+
+    // In the next available row, writes `a` into the row's left cell and `b` into the row's right
+    // cell.
+    fn alloc_private_inputs(
+        &self,
+        mut layouter: impl Layouter<Fp>,
+        a: Fp,
+        b: Fp,
+    ) -> Result<(Alloc, Alloc), Error> {
+        let config = self.config();
+        let mut a_alloc: Option<Alloc> = None;
+        let mut b_alloc: Option<Alloc> = None;
+
+        layouter.assign_region(
+            || "load private inputs",
+            |mut region| {
+                let row_offset = 0;
+                let a_cell = region.assign_advice(
+                    || "private input 'a'",
+                    config.l_col,
+                    row_offset,
+                    || Ok(a),
+                )?;
+                let b_cell = region.assign_advice(
+                    || "private input 'b'",
+                    config.r_col,
+                    row_offset,
+                    || Ok(b),
+                )?;
+                // Note that no arithmetic is performed here, all we are doing is allocating the
+                // initial private wire values (i.e. private values which are not the output of any
+                // gate), thus there is no selector enabled in this row.
+                a_alloc = Some(Alloc { cell: a_cell, value: a });
+                b_alloc = Some(Alloc { cell: b_cell, value: b });
+                Ok(())
```

Comment:
You can return variables from layouter regions, removing the need for this option pattern:
```suggestion
                Ok((Alloc { cell: a_cell, value: a }, Alloc { cell: b_cell, value: b }))
```